app/src/main/java/ch/epfl/sdp/peakar/social/SocialListAdapter.java

```diff
@@ -111,4 +115,17 @@ private void setMedal(ImageView v, int rank) {
                 break;
         }
     }
+
+    /**
+     * Set correct profile avatar
+     * @param v avatar view.
+     * @param avatarUrl avatar url of the social item.
+     */
+    private void setAvatar(ImageView v, Uri avatarUrl) {
+        if(avatarUrl == Uri.EMPTY) return;
+        Glide.with(mContext)
+                .load(avatarUrl)
+                .circleCrop()
```

Comment:
I know that with the gallery it is kind of jacked since there is a lot that is done in the getItem function from the listAdapter. It is called alot of times when scrolling etc. 
I don't know a lot about Glide if it is fast or slow. But would it be possible to do some of this on the social item instead, like everything before `.into(v)` when we create the item? 

So we don't have to do the load, crop etc. as much?

Reply:
I also have no experience with glide but it seemed really smooth to me on the phone and emulator.

Reply:
As @magangloff pointed out by trying on the phone, it should be smooth even if these lines were called many times.
In fact, when `Glide` downloads an image, it is stored in a cache. This way, even if the method is called multiple times the image is downloaded only once (as you can read [here](https://stackoverflow.com/questions/41758087/glide-not-updating-image-android-of-same-url)).
The other operations on the downloaded file should be very fast and I think we can easily ignore them.